src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -360,14 +375,22 @@ public void ExecuteSyntaxTreeActions(ImmutableArray<SyntaxTreeAnalyzerAction> sy
                 ExecuteSyntaxNodeActions(nodesToAnalyze, executableNodeActionsByKind, semanticModel, getKind);
             }
 
+            // Execute code block actions.
+            foreach (var blockAction in blockActions)
+            {
+                ExecuteAndCatchIfThrows(blockAction.Analyzer,
+                    () => blockAction.Action(new CodeBlockAnalysisContext(declaredNode, declaredSymbol, semanticModel, _analyzerOptions, _addDiagnostic, _cancellationToken)));
+            }
+
             // Execute code block end actions.
-            foreach (var endedAction in endedActions)
+            foreach (var endedAction in blockEndActions)
```

Comment:
This foreach loop seems identical to the one just above, probably can refactor it out to a method?